### DISCUSSION TITLE
how to setup consulo for unity game development ?

### DISCUSSION THREAD

**Discussion by coderhasib** [consulo/consulo#532] [General] [answered]
I have attached cosulo.sh as external editor in unity preference. But scripts not open with consulo. It open when press Assets>Open C# project. How to setup ? Thanks

**Comment by VISTALL** [comment]
Hello. https://github.com/consulo/UnityEditorConsuloPlugin required

**Comment by coderhasib** [comment]
how to install it?

**Reply by VISTALL** [reply]
You need add 

```
    "com.consulo.ide": "https://github.com/consulo/UnityEditorConsuloPlugin.git#2.6.0"
```

to Unity manifest file (there info about it in readme).

Also check https://consulo.help/plugins/consulo.unity3d/

**Comment by coderhasib** [comment]
is it possible this plugin manually ? like other pugin find and install in the consulo editor?

**Reply by VISTALL** [reply]
No. It's impossible installing Consulo plugin inside Unity without Unit Project modification.

You can copy code (from [UnityEditorConsuloPlugin](https://github.com/consulo/UnityEditorConsuloPlugin) to Unity project

**Comment by coderhasib** [comment]
unity package manger error: failed to resolve packages:.........
Not valid JSON: Unexpected string

**Reply by VISTALL** [reply]
Maybe you missed something - like comma?

**Comment by coderhasib** [comment]
An error occured while resolving packages:
Project has invalid dependencies:
com.consulo.ide: no git executabloe was found

**Reply by VISTALL** [reply]
Oh 'git' required by Unity (https://docs.unity3d.com/2023.2/Documentation/Manual/upm-ui-giturl.html)

**Comment by coderhasib** [comment]
i am very new so please dont mind for simple question. thanks for your help. i will try

**Comment by coderhasib** [comment]
i have installed git and UnityEditor Consulo Plugin but not open scripts  with consulo. I have reimport/all reimport also. but not work.

**Reply by VISTALL** [reply]
In Unity you have top menu like File Editor Help. You can find **Consulo** menu? Enable socket searching please

**Comment by coderhasib** [comment]
"Integration" and "Socket Searching " both active but not work. should i reboot?

**Reply by VISTALL** [reply]
Check Unity console - you have some errors? (Consulo opened? And installed Unity plugin inside Consulo)

**Reply by coderhasib** [reply]
everything looks good in unity no errors in console and no red mark in plugin section. but still not run scripts with consulo.

**Reply by VISTALL** [reply]
Can you show me Plugins list inside Consulo (File -> Settings -> Platform & Plugins)?

**Reply by coderhasib** [reply]
![Screenshot from 2023-08-13 06-25-33](https://github.com/consulo/consulo/assets/8093640/4afe38c4-5577-47de-9b17-88653c5b4130)
![Screenshot from 2023-08-13 06-25-39](https://github.com/consulo/consulo/assets/8093640/d04563f2-fdd9-454a-a3d6-c5293f633d4c)

i can open the scripts with consulo using select script and select open with c# project. only open one tab if i try to open another script.

**Reply by coderhasib** [reply]
how to change font size?

**Reply by coderhasib** [reply]
because it is inactive.

**Reply by coderhasib** [reply]
Consulo:
  version = 2023.7
  build number = 3355
  build date = July 17, 2023
 Plugins:
  com.intellij.git = 1719
  com.intellij.gwt = 14075
  com.intellij.images = 2738
  com.intellij.junit = 5034
  com.intellij.properties = 3271
  com.intellij.regexp = 1649
  com.intellij.spellchecker = 1666
  com.intellij.tasks = 5343
  com.intellij.uiDesigner = 6603
  com.intellij.velocity = 6828
  com.intellij.xml = 3194
  consulo = 3355
  consulo.apache.ant = 5283
  consulo.csharp = 6047
  consulo.desktop.awt = 3355
  consulo.devkit = 10783
  consulo.dotnet = 3935
  consulo.dotnet.core = 9100
  consulo.dotnet.microsoft = 4884
  consulo.dotnet.mono = 4895
  consulo.fontviewer = 242
  consulo.java = 5703
  consulo.javaee = 7750
  consulo.javascript = 3844
  consulo.jflex = 5057
  consulo.library.batik = 951
  consulo.library.flexmark = 951
  consulo.msbuild = 7202
  consulo.nuget = 5639
  consulo.nunit = 5637
  consulo.opencover = 4180
  consulo.testng = 5551
  consulo.unity3d = 9862
  org.intellij.groovy = 9530
  org.jetbrains.idea.maven = 6627
  org.jetbrains.plugins.github = 5154
  org.jetbrains.plugins.gradle = 11830
  org.jetbrains.plugins.yaml = 2595
 JVM:
  vendor = Private Build
  version = 17.0.8
  runtimeVersion = 17.0.8+7-Ubuntu-120.04.2
  locale = en_US
 JVM Env:
  awt.useSystemAAFontSettings = lcd
  consulo.home.path = /media/hasibul/WD_BACKUP/PROJECTS/IDE/inLinux/Consulo/platform/build3355
  consulo.laf.override.fonts = true
  consulo.module.path.boot = true
  consulo.properties.file = /media/hasibul/WD_BACKUP/PROJECTS/IDE/inLinux/Consulo/consulo.properties
  consulo.restart.code = 88
  consulo.vm.options.file = /media/hasibul/WD_BACKUP/PROJECTS/IDE/inLinux/Consulo/consulo64.vmoptions
  file.encoding = UTF-8
  file.separator = /
  idea.cycle.buffer.size = 1024
  idea.dynamic.classpath = false
  idea.home.path = /media/hasibul/WD_BACKUP/PROJECTS/IDE/inLinux/Consulo/platform/build3355
  idea.max.intellisense.filesize = 2500
  idea.no.launcher = false
  idea.popup.weight = heavy
  idea.properties.file = /media/hasibul/WD_BACKUP/PROJECTS/IDE/inLinux/Consulo/consulo.properties
  idea.use.default.antialiasing.in.editor = false
  io.netty.machineId = 28:f0:76:ff:fe:16:65:0e
  io.netty.processId = 11789
  io.netty.serviceThreadPrefix = Netty 
  java.class.path = 
  java.class.version = 61.0
  java.home = /usr/lib/jvm/java-17-openjdk-amd64
  java.io.tmpdir = /tmp
  java.library.path = /media/hasibul/WD_BACKUP/PROJECTS/IDE/inLinux/Consulo/platform/build3355/bin::/usr/java/packages/lib:/usr/lib/x86_64-linux-gnu/jni:/lib/x86_64-linux-gnu:/usr/lib/x86_64-linux-gnu:/usr/lib/jni:/lib:/usr/lib
  java.net.preferIPv4Stack = true
  java.rmi.server.disableHttp = true
  java.rmi.server.hostname = localhost
  java.runtime.name = OpenJDK Runtime Environment
  java.runtime.version = 17.0.8+7-Ubuntu-120.04.2
  java.security.manager = consulo.container.impl.securityManager.impl.ConsuloSecurityManager
  java.specification.name = Java Platform API Specification
  java.specification.vendor = Oracle Corporation
  java.specification.version = 17
  java.util.concurrent.ForkJoinPool.common.parallelism = 4
  java.util.concurrent.ForkJoinPool.common.threadFactory = consulo.bootstrap.concurrent.IdeaForkJoinWorkerThreadFactory
  java.vendor = Private Build
  java.vendor.url = Unknown
  java.vendor.url.bug = Unknown
  java.version = 17.0.8
  java.version.date = 2023-07-18
  java.vm.compressedOopsMode = 32-bit
  java.vm.info = mixed mode, sharing
  java.vm.name = OpenJDK 64-Bit Server VM
  java.vm.specification.name = Java Virtual Machine Specification
  java.vm.specification.vendor = Oracle Corporation
  java.vm.specification.version = 17
  java.vm.vendor = Private Build
  java.vm.version = 17.0.8+7-Ubuntu-120.04.2
  javax.swing.rebaseCssSizeMap = true
  jb.vmOptionsFile = /media/hasibul/WD_BACKUP/PROJECTS/IDE/inLinux/Consulo/consulo64.vmoptions
  jdk.debug = release
  jdk.module.main = consulo.desktop.awt.bootstrap
  jdk.module.main.class = consulo.desktop.awt.boot.main.Main
  jdk.module.path = /media/hasibul/WD_BACKUP/PROJECTS/IDE/inLinux/Consulo/platform/build3355/boot
  jna.loaded = true
  jna.nosys = true
  jna.platform.library.path = /usr/lib/x86_64-linux-gnu:/lib/x86_64-linux-gnu:/usr/lib64:/lib64:/usr/lib:/lib:/lib/i386-linux-gnu:/lib/i386-linux-gnu/i686/sse2:/lib/i386-linux-gnu/sse2:/usr/lib/x86_64-linux-gnu/libfakeroot
  jna.tmpdir = /home/hasibul/.consulo_settings/system/tmp
  jnidispatch.path = /home/hasibul/.consulo_settings/system/tmp/jna15705805056946017896.tmp
  line.separator = \n
  log4j.shutdownHookEnabled = false
  log4j2.messageFactory = org.apache.logging.log4j.message.SimpleMessageFactory
  native.encoding = UTF-8
  os.arch = amd64
  os.name = Linux
  os.version = 5.15.0-78-generic
  path.separator = :
  sun.arch.data.model = 64
  sun.awt.datatransfer.timeout = 2000
  sun.awt.enableExtraMouseButtons = true
  sun.awt.exception.handler = consulo.desktop.awt.application.impl.AWTExceptionHandler
  sun.awt.noerasebackground = true
  sun.boot.library.path = /usr/lib/jvm/java-17-openjdk-amd64/lib
  sun.cpu.endian = little
  sun.font.fontmanager = sun.awt.X11FontManager
  sun.io.unicode.encoding = UnicodeLittle
  sun.io.useCanonCaches = false
  sun.java.command = consulo.desktop.awt.bootstrap/consulo.desktop.awt.boot.main.Main
  sun.java.launcher = SUN_STANDARD
  sun.java2d.d3d = false
  sun.java2d.noddraw = true
  sun.java2d.pmoffscreen = false
  sun.java2d.uiScale.enabled = true
  sun.jnu.encoding = UTF-8
  sun.management.compiler = HotSpot 64-Bit Tiered Compilers
  sun.nio.ch.bugLevel = 
  swing.bufferPerWindow = false
  user.country = US
  user.dir = /media/hasibul/WD_BACKUP/PROJECTS/All_Projects/Unity Projects/Prototype_3D_03
  user.home = /home/hasibul
  user.language = en
  user.name = hasibul
  user.timezone = Asia/Dhaka
 OS:
  name = Linux
  version = 5.15.0-78-generic
  arch = amd64
 Env:
  ALL_JVM_ARGS = -Djava.security.manager=consulo.container.impl.securityManager.impl.ConsuloSecurityManager --add-opens=java.desktop/sun.awt=consulo.container.impl --add-opens=java.desktop/sun.swing=consulo.container.impl --add-opens=java.desktop/sun.awt.image=consulo.container.impl --add-opens=java.desktop/sun.java2d=consulo.container.impl --add-opens=java.desktop/sun.font=consulo.container.impl --add-opens=java.desktop/java.awt=consulo.container.impl --add-opens=java.desktop/javax.swing=consulo.container.impl --add-opens=java.desktop/javax.swing.plaf.basic=consulo.container.impl --add-opens=java.base/java.lang=consulo.container.impl --add-opens=java.desktop/javax.swing.text.html=consulo.container.impl --add-opens=java.desktop/java.awt.peer=consulo.container.impl --add-opens=java.desktop/java.awt.event=consulo.container.impl --add-opens=java.desktop/com.apple.laf=consulo.container.impl --add-opens=java.desktop/com.apple.eawt=consulo.container.impl --add-opens=java.desktop/com.apple.eawt.event=consulo.container.impl  -Xms512m -Xmx1500m -XX:ReservedCodeCacheSize=240m -ea -Dsun.io.useCanonCaches=false -Djava.net.preferIPv4Stack=true -XX:+UseCodeCacheFlushing -XX:SoftRefLRUPolicyMSPerMB=50 -XX:+HeapDumpOnOutOfMemoryError -Dawt.useSystemAAFontSettings=lcd -Dconsulo.laf.override.fonts=true  -Djb.vmOptionsFile="/media/hasibul/WD_BACKUP/PROJECTS/IDE/inLinux/Consulo/consulo64.vmoptions" -Dconsulo.vm.options.file="/media/hasibul/WD_BACKUP/PROJECTS/IDE/inLinux/Consulo/consulo64.vmoptions" -p "/media/hasibul/WD_BACKUP/PROJECTS/IDE/inLinux/Consulo/platform/build3355/boot" -Dconsulo.module.path.boot=true -Dconsulo.home.path="/media/hasibul/WD_BACKUP/PROJECTS/IDE/inLinux/Consulo/platform/build3355" -Didea.home.path="/media/hasibul/WD_BACKUP/PROJECTS/IDE/inLinux/Consulo/platform/build3355" -Didea.properties.file="/media/hasibul/WD_BACKUP/PROJECTS/IDE/inLinux/Consulo/consulo.properties" -Dconsulo.properties.file="/media/hasibul/WD_BACKUP/PROJECTS/IDE/inLinux/Consulo/consulo.properties"   
  AMPLITUDE_API_KEY = 28f3cae0b33b8b3702120c9ed1a935aa
  AMPLITUDE_API_KEY_STAGING = 83ee04fccc67e4fcc7dd527c4c6f1e21
  CHROME_DESKTOP = unityhub.desktop
  CONSULO_HOME = /media/hasibul/WD_BACKUP/PROJECTS/IDE/inLinux/Consulo/platform/build3355
  DBUS_SESSION_BUS_ADDRESS = unix:path=/run/user/1000/bus
  DEBUG_ENV_VAR = UnityHub
  DESKTOP_SESSION = zorin
  DISPLAY = :1
  GDK_BACKEND = x11
  GDMSESSION = zorin
  GIO_LAUNCHED_DESKTOP_FILE = /usr/share/applications/unityhub.desktop
  GIO_LAUNCHED_DESKTOP_FILE_PID = 8466
  GNOME_DESKTOP_SESSION_ID = this-is-deprecated
  GNOME_SHELL_SESSION_MODE = zorin
  GPG_AGENT_INFO = /run/user/1000/gnupg/S.gpg-agent:0:1
  GSETTINGS_SCHEMA_DIR = /home/hasibul/data
  GTK_MODULES = gail:atk-bridge
  HEAP_CLIENT_TOKEN = 1717415662
  HEAP_CLIENT_TOKEN_STAGING = 3786794399
  HOME = /home/hasibul
  IM_CONFIG_PHASE = 1
  INVOCATION_ID = abd7628ac1c44c67ae11fa1d0eb50c13
  JDK = /usr/lib/jvm/java-17-openjdk-amd64
  JOURNAL_STREAM = 8:42716
  LANG = en_US.UTF-8
  LAUNCH_DARKLY_CLIENT_TOKEN = 60a7e30fafc7e60e33446283
  LAUNCH_DARKLY_CLIENT_TOKEN_STAGING = 60a7e30fafc7e60e33446282
  LC_ADDRESS = bn_BD
  LC_IDENTIFICATION = bn_BD
  LC_MEASUREMENT = bn_BD
  LC_MONETARY = bn_BD
  LC_NAME = bn_BD
  LC_NUMERIC = bn_BD
  LC_PAPER = bn_BD
  LC_TELEPHONE = bn_BD
  LC_TIME = bn_BD
  LD_LIBRARY_PATH = /media/hasibul/WD_BACKUP/PROJECTS/IDE/inLinux/Consulo/platform/build3355/bin:
  LOGNAME = hasibul
  MAIN_CLASS_NAME = -m consulo.desktop.awt.bootstrap/consulo.desktop.awt.boot.main.Main
  MANAGERPID = 2336
  NO_AT_BRIDGE = 1
  ORIGINAL_XDG_CURRENT_DESKTOP = zorin:GNOME
  PATH = /usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin:/usr/games:/usr/local/games:/snap/bin:/snap/bin
  PWD = /media/hasibul/WD_BACKUP/PROJECTS/All_Projects/Unity Projects/Prototype_3D_03
  QT_ACCESSIBILITY = 1
  QT_IM_MODULE = ibus
  ROOT_DIR = /media/hasibul/WD_BACKUP/PROJECTS/IDE/inLinux/Consulo
  SENTRY_ACCESS_TOKEN = 3df78d9e53f8456aa90fbae044ce1a6261421ab1d0014674bc2e933477289017
  SESSION_MANAGER = local/hasibul-S551LN:@/tmp/.ICE-unix/2603,unix/hasibul-S551LN:/tmp/.ICE-unix/2603
  SHELL = /bin/bash
  SHLVL = 0
  SSH_AGENT_PID = 2566
  SSH_AUTH_SOCK = /run/user/1000/keyring/ssh
  USER = hasibul
  USERNAME = hasibul
  WINDOWPATH = 2
  WOOTRIC_CLIENT_TOKEN = NPS-aa8be4c3
  XAUTHORITY = /run/user/1000/gdm/Xauthority
  XDG_CONFIG_DIRS = /etc/xdg/xdg-zorin:/etc/xdg
  XDG_CURRENT_DESKTOP = zorin:GNOME
  XDG_DATA_DIRS = /usr/share/zorin:/home/hasibul/.local/share/flatpak/exports/share:/var/lib/flatpak/exports/share:/usr/local/share/:/usr/share/:/var/lib/snapd/desktop
  XDG_MENU_PREFIX = gnome-
  XDG_RUNTIME_DIR = /run/user/1000
  XDG_SESSION_CLASS = user
  XDG_SESSION_DESKTOP = zorin
  XDG_SESSION_TYPE = x11
  XMODIFIERS = @im=ibus
  __GL_SYNC_TO_VBLANK = 0

**Reply by VISTALL** [reply]
Hmm. Unity installed. Can you do this:

* Open Unity
* Open Consulo
* Double click inside Unity on C# file
* Go to Consulo. 
* Open Help -> Show Log in *** (name of your file manager), and attach consulo.log file here

**Reply by VISTALL** [reply]
Tested - looks like something is broken, working on it. Sorry, and thanks for reporting (hard to test all platforms)

**Comment by VISTALL** [comment, marked as answer]
Hello. I have update. I fixed all issues which I catch.

Please do this actions:
* File -> Settings -> Platform & Plugins -> Update Settings
* Change **release** update channel to **nightly**
* Press OK (close Settings dialog)
* Help -> Check for Updates
* Press OK (and wait until all finished, and restart Consulo)
* 
Sorry for issues

**Reply by coderhasib** [reply]
Thanks for your good work.

**Reply by coderhasib** [reply]
would you please tell me how to change font size? because all options are disabled.

**Reply by VISTALL** [reply]
Default color schemes readonly, press **Save as...** button - and editing will be available

**Reply by coderhasib** [reply]
Thanks. it works.

**Comment by coderhasib** [comment]
All are working now. Thanks for your help.

**Reply by VISTALL** [reply]
Nice, thanks for reports, and sorry

**Comment by coderhasib** [comment]
Following plugins are incompatible with current IDE build: Git Integration, Regexp, Simple Spellchecker, XML/HTML, Properties Support, .NET Common, .NET Framework, Mono (.NET Framework-compatible framework), C#, JavaScript, Batik Library, Images, Java, JUnit, Swing UI Designer, Apache Velocity, Apache Ant, JFlex Support, FlexMark Library, Common Task Servers, MSBuild, .NET (previously named .NET Core), NuGet, NUnit, OpenCover, Python, TestNG-J, Maven Integration, Jakarta EE (Java EE), GWT, Groovy, GitHub, Gradle, JavaFX, Consulo Plugin DevKit, IntelliJ Project Importer, Unity (Game Engine)

after update. please give latest version for zorin linux.

**Reply by VISTALL** [reply]
Hello. Before updating - Consulo give you warning

![image](https://github.com/consulo/consulo/assets/542934/771ac9db-39f0-4cd3-bfc5-7c25527b0589)

Looks like you fully ignore it. Please wait until new update ready - few hours need

**Comment by coderhasib** [comment]
ok. waiting ....

**Comment by coderhasib** [comment]
show "Checksum mismatch. Try again later. " after finishing the update.

**Reply by VISTALL** [reply]
Try again now/later - it's plugin update uploaded in same time, and checksum has been changed on server (versions not stable - just wait), and can try update again - after sometime, will be ok

**Reply by VISTALL** [reply]
Now must be ok

**Comment by coderhasib** [comment]
Thanks. consulo editor not open double click on the unity c# file.

**Reply by VISTALL** [reply]
Depends on Consulo/Socket Searching option. (better use Desktop Entry from Unity). Can you close Consulo, and rerun it from menu?

**Comment by coderhasib** [comment]
consulo ide is free for forever or will be paid like jetbrain rider?

**Reply by VISTALL** [reply]
Fully free, fully opensourced.

**Comment by coderhasib** [comment]
Thanks, man for your hardy nicy work. appreciate it. I have waited like this project. Thanks again from my heart for your quick response also. will pray for you inshallah.

**Reply by VISTALL** [reply]
Please be careful while updating. Better - change update settings to **release**.

I'm hard working on stability of Linux/Mac versions.

**Comment by coderhasib** [comment]
I will do it and be careful about it Inshallah. one more thing. failed to create a desktop entry.

**Comment by coderhasib** [comment]
"Can not find file ' /file path' " when trying to open c# file from unity.

**Reply by VISTALL** [reply]
Hello. It's real message? or you make it shorten? That issue you already catch before (but there was error in console) - I need full message for understanding problem (better screenshot), since I'm not sure who broke file path

**Comment by coderhasib** [comment]
Hi there, sorry for late and yes I make it shorter. Here is the main error message.
![Screenshot from 2023-08-30 10-25-34](https://github.com/consulo/consulo/assets/8093640/8825f8d1-0d3c-48cc-b5de-eab09a3874fe)
![Screenshot from 2023-08-30 10-05-03](https://github.com/consulo/consulo/assets/8093640/be5c1a31-1c50-4ff3-aef4-ed8d7da6d7af)

**Reply by VISTALL** [reply]
Hmm. It's removable device? Also it's strange path must be pointed to file, not a directory

**Comment by coderhasib** [comment]
Yes you are right. This is my backup drive which is attachted inside my laptop.So, what is the solution bro.

**Reply by VISTALL** [reply]
For now I don't have an idea for why that happens. I will investigate that  (review project opening code)

**Reply by coderhasib** [reply]
okay. Thanks for your co-operation.

**Reply by coderhasib** [reply]
Hello, Have you finished your investigation? any alternative ways?

**Reply by VISTALL** [reply]
Hello. For now - no. Looks like for me - Unity ignored plugin. I'm not sure what Unity send to Consulo. 

Can you add custom define CONSULO_ACTION_DEBUG  (https://docs.unity3d.com/Manual/CustomScriptingSymbols.html) - and watch what Unity write in console while double click on file?

**Reply by coderhasib** [reply]
I have downloaded the new version and removed the old consolu config file from the home directory reinstalled everything and updated the unity plugin, it sometimes works but not all the time. This message gives sudden. So, I can not work.

**Comment by coderhasib2021** [comment]
Hello VISTALL, Thanks once again for your wonderful work. I am facing this problems now. 1. currently I am using 

Consulo:
  version = 2023.12
  build number = 3412
  build date = December 18, 2023

but it is not open from unity editor automatically after double click on script  or from open C# project.
![IMG_20240911_115708](https://github.com/user-attachments/assets/599d60f8-dd15-4628-ac0f-47fff53e1641)
![Screenshot from 2024-09-11 11-54-57](https://github.com/user-attachments/assets/c52f6119-bc91-43f1-9163-d00c06b4a5fe)

problem 2: I have tried to update several times but get this err: 
![Screenshot from 2024-09-11 11-13-28](https://github.com/user-attachments/assets/570d4cfa-9884-46bd-af61-5f92591fd4ba)

Please fix this as soon as possible. thanks again for your good work and quick response. I really love your project.

**Comment by VISTALL** [comment]
Hello. can you try open this url in browser? https://api.consulo.io/repository/list?channel=nightly&platformVersion=SNAPSHOT

There no problem at webserver - I think it's from with your side, but I will try to find the way for fix

**Comment by coderhasib2021** [comment]
![Screenshot from 2025-03-05 22-31-04](https://github.com/user-attachments/assets/52cebca1-7120-4d76-a5f7-fc1805acf5e4)
need help to install. thanks

**Reply by VISTALL** [reply]
Hello. Please say version from Help -> About? This plugin looks like installed

**Comment by coderhasib2021** [comment]
version = 2025.3

**Reply by VISTALL** [reply]
and build? What problem you got?  You can install simple by this steps. 

![image](https://github.com/user-attachments/assets/869a9f18-3603-44e4-ab33-a9bb7170d188)

Also - Check for Updates call required

**Reply by coderhasib2021** [reply]
Plugin "Game Framework API" was not loaded: required plugin "consulo.audio" not installed. Plugin "Unity (Game Engine)" was not loaded: required plugin "consulo.audio" not installed.

**Reply by VISTALL** [reply]
Find Audio plugin - and install it in Repository list

**Reply by VISTALL** [reply]
![{532FCB4C-12FD-409B-8291-8DEEFA0A27AB}](https://github.com/user-attachments/assets/716c193c-dbe9-4e3c-b0de-0755a3ea1a10)

**Reply by coderhasib2021** [reply]
Thanks problem fixed. if face same problem next time what i will do? and why show this? i have tried many times and did not work but now worked. how to backup and restore settings

**Reply by VISTALL** [reply]
Looks like new dependency not downloaded. I will check why its not downloaded.

It's just a bug. Audio plugin and Game Framework API its new plugins, and Unity depends on it. 

Plugins have hard dependency to new Consulo version, thats why no sense store them(export) and later insert it to new Consulo.

Consulo settings (like editor options) - can be exported via `File -> Export Settings`
